File: backend/test/c/test_typedefs.c
#include <assert.h>

#include "test_typedefs_c.h"

int main()
{
  {
    bool_type_t b = 1;
    bool_type_t not_b = test_logical_not(b);
    assert(not_b == 0);
  }

  {
    int_type_1_t a = 1;
    int_type_2_t b = 2;
    test_int_type_common_t c = test_add(a, b);
    assert(c == 3);
  }

  {
    test_s_t s = test_s_new();
    test_s_t s_copy = test_copy(&s);

    test_s_delete(&s);
    test_s_delete(&s_copy);
  }

  return 0;
}

File: backend/test/rust/test_typedefs.rs
use std::ffi::*;

include!("test_typedefs_rust.rs");

fn main() {
    unsafe {

    {
      let b: BoolType = true;
      let not_b: BoolType = test_logical_not(b);
      assert_eq!(not_b, false);
    }

    {
      let a: IntType1 = 1;
      let b: Scope1Scope2IntType2 = 2;
      let c: TestIntTypeCommon = test_add(a, b);
      assert_eq!(c, 3);
    }

    {
      let s: TestST = TestS::new();
      let s_ptr: TestSPtrT = &s;
      let _s_copy: TestST = test_copy_s_ptr(s_ptr);
    }

    }
}
